### DISCUSSION TITLE
Combine multi query data?

### DISCUSSION THREAD

**Discussion by bebiangel** [TanStack/query#2374] [Q&A]
Hi, I am asking because I need answers and opinions while using react query.

The first question was trying to format the data by calling another api after **onSuccess** in the link below.
However, if I remove the comment, it makes an infinite call, what is the cause?
 
The second question requires an idea.
Similar to the above question, I want to make a query hook that combines data by making consecutive asynchronous calls.
Once you look at the source, I used useState to combine data, but I think there are better opinions.

If you have any ideas, I'd appreciate it if you could give me a comment.
Thank you!

https://codesandbox.io/s/set-query-test-qzhl0?file=/src/List.js

**Comment by TkDodo** [comment]
> what is the cause?

setQueryData calls onSuccess, that’s the cause. 

Have you looked at dependent queries: https://react-query.tanstack.com/guides/dependent-queries
